### PR TITLE
fix: Add check for ticket and CFS ends-at

### PR DESCRIPTION
**Pull request by shreyanshdwivedi**
Fixes #
Related issue https://github.com/fossasia/open-event-frontend/issues/2653

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

- [ ] All the functions created/modified in this PR contain relevant docstrings.

#### Short description of what this resolves:
Currently, the CFS end date and ticket end date can be more than the event end date. Proper error must be raised on such cases

#### Changes proposed in this pull request:
- Add check for these conditions

### CODE REVIEW

File: app/api/schema/tickets.py [hunk 1]
```diff
@@ -29,11 +29,22 @@ def validate_date(self, data, original_data):
...
                 data['sales_ends_at'] = ticket.sales_ends_at
+         
+            if 'event_ends_at' not in data:
+                data['event_ends_at'] = ticket.event.ends_at
 
         if data['sales_starts_at'] >= data['sales_ends_at']:
             raise UnprocessableEntity({'pointer': '/data/attributes/sales-ends-at'},
                                       "sales-ends-at should be after sales-starts-at")
 
+        if data['sales_starts_at'] > data['event_ends_at']:
```
Comment (uds5501) [inline app/api/schema/tickets.py]: Put these  in two conditions in
```python

if event_ends_at in data

```
maybe this is the reason of validation error?

Comment (shreyanshdwivedi) [reply app/api/schema/tickets.py]: Thanks @uds5501. It helped :)

**Comment by shreyanshdwivedi** [comment]
@iamareebjamal @CosmicCoder96 please review :)

File: app/api/schema/speakers_calls.py [hunk 2]
```diff
@@ -32,9 +32,20 @@ def validate_date(self, data, original_data):
...
         if data['starts_at'] >= data['ends_at']:
             raise UnprocessableEntity({'pointer': '/data/attributes/ends-at'}, "ends-at should be after starts-at")
 
+        if data['starts_at'] > data['event_ends_at']:
+            raise UnprocessableEntity({'pointer': '/data/attributes/starts-at'},
+                                      "speakers-call starts-at should be before event ends-at")
+
+        if data['ends_at'] > data['event_ends_at']:
+            raise UnprocessableEntity({'pointer': '/data/attributes/ends-at'},
+                                      "speakers-call ends-at should be before event ends-at")
```
Comment (iamareebjamal) [inline app/api/schema/speakers_calls.py]: Shouldn't it be before starts-at?

Comment (shreyanshdwivedi) [reply app/api/schema/speakers_calls.py]: @iamareebjamal for CFS, I think starts-at should be used. Thanks for pointing out, I used the same condition as for tickets here :sweat_smile: 
Will be making the changes :)

Comment (shreyanshdwivedi) [reply app/api/schema/speakers_calls.py]: @iamareebjamal I've made the required changes. Can you please transfer this issue to server - https://github.com/fossasia/open-event-frontend/issues/2653 ?
Also, please review

**Comment by shreyanshdwivedi** [comment]
@CosmicCoder96 @iamareebjamal please review

**Comment by iamareebjamal** [comment]
Please everyone, add a semantic PR title so our bot can label the PR correctly which will make it easy for us to generate release notes

**Pull request merged**